applications/MultilevelMonteCarloApplication/python_scripts/adaptive_refinement_utilities.py

```diff
@@ -15,65 +19,135 @@
     exploiting the hessian metric of the solution
     input:  self: an instance of the class
-    output: current_model_refined:      Kratos model class after refinement
-            current_parameters_refined: Kratos parameters class after refinement
+    output: current_model_refined      : Kratos model class after refinement
+            current_parameters_refined : Kratos parameters class after refinement
     """
     def ComputeAdaptiveRefinement(self):
         parameters_coarse = self.parameters_coarse
         model_coarse = self.model_coarse
-        minimal_size_value = self.minimal_size
-        maximal_size_value = self.maximal_size
         metric_param = self.metric_param
         remesh_param = self.remesh_param
         problem_type = self.problem_type
+        current_level = self.current_level
 
         if (self.metric is "hessian"):
-            model_part_name = parameters_coarse["problem_data"]["model_part_name"].GetString()
-            # set NODAL_AREA and NODAL_H as non historical variables
-            KratosMultiphysics.VariableUtils().SetNonHistoricalVariable(KratosMultiphysics.NODAL_AREA, 0.0, model_coarse.GetModelPart(model_part_name).Nodes)
-            KratosMultiphysics.VariableUtils().SetNonHistoricalVariable(KratosMultiphysics.NODAL_H, 0.0, model_coarse.GetModelPart(model_part_name).Nodes)
-            # calculate NODAL_H
-            find_nodal_h = KratosMultiphysics.FindNodalHProcess(model_coarse.GetModelPart(model_part_name))
-            find_nodal_h = KratosMultiphysics.FindNodalHNonHistoricalProcess(model_coarse.GetModelPart(model_part_name))
-            find_nodal_h.Execute()
-            # prepare parameters to calculate the gradient of the designed variable
-            local_gradient_variable_string = metric_param["local_gradient_variable"].GetString()
-            local_gradient_variable = KratosMultiphysics.KratosGlobals.GetVariable(metric_param["local_gradient_variable"].GetString())
-            metric_param.RemoveValue("local_gradient_variable")
-            metric_param.AddEmptyValue("minimal_size")
-            metric_param["minimal_size"].SetDouble(minimal_size_value)
-            metric_param.AddEmptyValue("maximal_size")
-            metric_param["maximal_size"].SetDouble(maximal_size_value)
-            # calculate the gradient of the variable
-            local_gradient = KratosMeshing.ComputeHessianSolMetricProcess(model_coarse.GetModelPart(model_part_name),local_gradient_variable,metric_param)
-            local_gradient.Execute()
-            # add again the removed variable parameter
-            metric_param.AddEmptyValue("local_gradient_variable")
-            metric_param["local_gradient_variable"].SetString(local_gradient_variable_string)
+            original_interp_error = metric_param["hessian_strategy_parameters"]["interpolation_error"].GetDouble()
+
+            # problem dependent section
+            if (problem_type == "body_fitted_ellipse"):
```

Comment:
Here we are using `parameters["problem_data"]["problem_name"]` to distinguish between different cases, which is a user-defined name.

I would expect this to be something like "problem_type = potential_flow" or "problem_type = fluid_dynamics", something that is added in the parameters_refinement.

Reply:
Done!